ros_ws/src/dataset_generation/include/dataset_generation/filter_point_clouds.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <dirent.h>
+#include <iostream>
+#include <thread>
+#include <string>
+#include <sstream>
+#include <unordered_set>
+
+#include <pcl/ModelCoefficients.h>
+#include <pcl/PCLPointCloud2.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/common/io.h>
+#include <pcl/point_types.h>
+#include <pcl/common/common_headers.h>
+#include <pcl/filters/voxel_grid.h>
+
+#include "extract_point_cloud_objects.h"
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include "dataset_generation/json.hpp"
+
+using namespace std::literals::chrono_literals;
+
+namespace dataset_generation
+{
+
+
+class PCDFiltering
+{
+public:
+    PCDFiltering(
+        std::string in_folder_path,
+        std::string object_name,
+        std::string out_folder_path);
+
+    ~PCDFiltering();
+
+    void VoxelFilter(float leaf_size_x, float leaf_size_y, float leaf_size_z);
+private:
+    std::string in_folder_path_;
+    std::string object_name_;
+    std::string out_folder_path_;
+};
+PCDFiltering::PCDFiltering(
+        std::string in_folder_path,
+        std::string object_name,
+        std::string out_folder_path)
+{
+    in_folder_path_=in_folder_path;
+    object_name_=object_name;
+    out_folder_path_=out_folder_path;
+}
+PCDFiltering::~PCDFiltering()
+{}
+
+/*
+This method is a modified version of the VoxelGrid filtering found at: 
+https://pcl-tutorials.readthedocs.io/en/master/voxel_grid.html?highlight=voxelGrid
+*/
+void PCDFiltering::VoxelFilter(float leaf_size_x, float leaf_size_y, float leaf_size_z)
+{
+    pcl::PCLPointCloud2::Ptr cloud_read (new pcl::PCLPointCloud2 ());
+    pcl::PCLPointCloud2::Ptr cloud_voxelized (new pcl::PCLPointCloud2 ());
+    
+    //Read PCD files from the path given
+    std::string path=in_folder_path_+object_name_+".pcd";
```

Comment:
This path is not generalized. The user should be able to supply any Point Cloud to filter.

Reply:
Then I would just be able to read the string and load the PCD file specified at the end of the string.